### DISCUSSION TITLE
Securing the Pocketbase API

### DISCUSSION THREAD

**Discussion by TannerGabriel** [pocketbase/pocketbase#672] [Q&A] [answered]
I followed the documentation, successfully set up a Pocketbase instance, and implemented authentication in my Nuxt2 application, which worked great. The only question that remains is how the Pocketbase API itself is secured. After following the documentation, anybody can open a connection to the Pocketbase API using just the `Base URL` with no API token involved.

```javascript
const client = new PocketBase("http://127.0.0.1:8090");
```

Is there any built-in functionality to secure the instance with an API token or something similar (like Firebase or Supabase)? If not, is there an example of how to secure the API  (reverse proxy with authentication, etc.) or any best practice guidelines?

**Comment by ganigeorgiev** [comment, marked as answer]
Access control of your data in PocketBase is handled via the collection API rules. Currently the documentation is a bit lacking, but you could find some info in https://pocketbase.io/docs/manage-collections/#collection-api-rules.

---

> Is there any built-in functionality to secure the instance with an API token or something similar?

API token for what? For authentication, you could check https://pocketbase.io/docs/api-authentication/.

**Reply by ganigeorgiev** [reply]
_I'm not sure if this is what you are asking, but if you want to instruct the browsers to allow only your domain to be able to make requests to your backend, you can enable cors by starting the executable with `--origins=mydomain1.com,mydomain2.com` flag._

**Reply by TannerGabriel** [reply]
Thank you for the quick reply. I have published my Pocketbase instance on a public domain and wanted to ensure that no unauthenticated person can read and write to my databases. The first option with the API rules worked perfectly for that.
